invenio_stats/aggregations.py

```diff
@@ -97,13 +99,26 @@ def wrapped(self, *args, **kwargs):
 
         return wrapped
 
+    def prepare_bookmark(self, value=None):
+        """Prepare the value that will be set by the bookmark."""
+        if value:
+            self.new_timestamp = value
+        else:
+            # By default, it is the current timestamp, minus 1 minute, since indices might take a bit to be refreshed
+            self.new_timestamp = (datetime.utcnow() - timedelta(minutes=1)).strftime(
+                "%Y-%m-%dT%H:%M:%S"
+            )
+
     @_ensure_index_exists
-    def set_bookmark(self, value):
+    def commit_bookmark(self):
```

Comment:
for my personal info: what is the difference between set and commit ?

Reply:
THe idea that I had ws to mimic a transaction in a database. First, the "prepare_bookmark" is called when the collector is about to start. That sets already the time, although the time is not written until the `commit`. Note that the signature between set and commit has changed

Reply:
I see. I would decouple the logic of stats and collector. It just happens that InvenioRDM is using stats in a specific way, and that it requires to re-index records. Other instances using `invenio-stats` might not require that at all.
I think that we should not add logic specific to/because of InvenioRDM.

Reply:
Ok, the signature of the document has been moved back to its former state. For the record, this thing of the `commit` was not caused by the dependency of invenioRDM. The same issue exists for the stats collectors used here (see the comment of the 1 minute delay